pkg/monitor/monitor/monitor_manager.go

```diff
@@ -58,6 +60,11 @@ func NewMonitorManager(deps *controller.Dependencies) *MonitorManager {
 }
 
 func (m *MonitorManager) SyncMonitor(monitor *v1alpha1.TidbMonitor) error {
+	// syncing all PVs managed by operator's reclaim policy to Retain
+	if err := m.pvManager.SyncMonitor(monitor); err != nil {
+		return err
+	}
```

Comment:
We can move this block to the code where the PV labels are synced as they are both for PVs.